addons/azureSets/azuresets.lua

```diff
@@ -133,15 +182,16 @@ function get_current_spellset()
     local spellTable = T{}
     local tmpTable = T{}
     if windower.ffxi.get_player()['main_job_id'] == 16 then
-        local tmpTable = T(windower.ffxi.get_mjob_data()['spells'])
-        local i,id
-        for i = 1, #tmpTable do
-            local t = ''
-            if tonumber(tmpTable[i]) ~= 512 then
+        local tmp = windower.ffxi.get_mjob_data()['spells']
+        for k,v in pairs(tmp) do
+            if tonumber(v) ~= 512 then
                 for spell in spells:it() do
```

Comment:
While not technically wrong, I don't know why it iterates the spells to find the one with the right ID... in fact, this entire function does a lot of unnecessary boilerplate to get its result. It can be replaced entirely with this:
```lua
function get_current_spellset()
    if windower.ffxi.get_player().main_job_id ~= 16 then return nil end
    return T(windower.ffxi.get_mjob_data().spells)
        -- Returns all values but 512
        :filter(function(id) return id ~= 512 end)
        -- Transforms them from IDs to lowercase English names
        :map(function(id) return spells[id].english:lower() end)
        -- Transform the keys from numeric x or xx to string 'slot0x' or 'slotxx'
        :keymap(function(slot) return '%02u':format(slot) end)
end
```

Learn functional programming people, will make your lives easier :)